internal/experiment/generator.go

```diff
@@ -172,20 +193,20 @@ func (g *Generator) selectors() []scan.Selector {
 	// TODO IngressSelector
 	// TODO ConfigMapSelector?
 
-	// The application selector runs last and looks at the application configuration itself
-	result = append(result, &generation.ApplicationSelector{
-		Application: &g.Application,
-	})
-
 	return result
 }
 
 // validate is basically just a hook to perform final verifications before actually emitting anything.
 func (g *Generator) validate([]*yaml.RNode) error {
 
-	for i := range g.Application.Objectives {
-		if !g.Application.Objectives[i].Implemented {
-			return fmt.Errorf("generated experiment cannot optimize objective: %s", g.Application.Objectives[i].Name)
+	objective, err := application.GetObjective(&g.Application, g.Objective)
+	if err != nil {
+		return err
+	}
+
+	for i := range objective.Goals {
```

Comment:
Wont this be a sad day if objective == nil?

Reply:
A sad day indeed.